dash/dash.py

```diff
@@ -475,6 +475,110 @@ def _validate_callback(self, output, inputs, state, events):
                 output.component_id,
                 output.component_property).replace('    ', ''))
 
+    def _validate_callback_output(self, output_value, output):
+        valid = [str, dict, int, float, type(None), Component]
```

Comment:
Since we JSON serialize with the `json.dumps(obj, cls=plotly.utils.PlotlyJSONEncoder)`, this list actually has a few other items: https://github.com/plotly/plotly.py/blob/6b3a0135977b92b3f7e0be2a1e8b418d995c70e3/plotly/utils.py#L137-L332

So, if there was a component property that accepted a list of numbers, then technically the user could return a numpy array or a dataframe.

The only example that immediately comes to mind is updating the `figure` property in a callback with the `plotly.graph_objs`. These objects get serialized because they have a `to_plotly_json` method.

So, I wonder if instead of validating up-front, we should only run this routine only if our `json.dumps(resp, plotly.utils.PlotlyJSONEncoder)` call fails.

This would have the other advantage of being faster for the non-error case. If the user returns a huge object (e.g. some of my callbacks in apps return a 2MB nested component), doing this recursive validation might be slow.

Reply:
That makes sense. I am not sure how large of a speed up that would be since validating a large callback output is likely much faster than the network delay of sending that large output, but it definitely makes it easier than crafting a perfect validator (that would need to update every time we want to extend `PlotlyJSONEncoder`).